File: 프로그래머스/lv2/42586. 기능개발/기능개발.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(vector<int> progresses, vector<int> speeds) {
    vector<int> answer,days;
    int cnt=1;
    for(int i=0;i<progresses.size();i++){
        int res=100-progresses[i];
        if(res%speeds[i]){
            days.push_back(res/speeds[i]+1);
        }
        else days.push_back(res/speeds[i]);
    }
    
    while(!days.empty()){
        cnt=1;
        int cur=days.front();
        for(int i=1;i<days.size();i++){
            if(cur>=days[i]){
                cnt++;
            }
            else break;
        }
        for(int i=0;i<cnt;i++){
            days.erase(days.begin());
        }
        answer.emplace_back(cnt);
    }
    
    
    return answer;
}

File: 프로그래머스/2/42586. 기능개발/기능개발.java
import java.util.*;

class Solution {
    public int[] solution(int[] progresses, int[] speeds) {
        List<Integer> answer = new ArrayList<>();
        int[] works = new int[progresses.length];

        for(int i=0; i<progresses.length; i++){
            works[i] = (int) (100-progresses[i]) / speeds[i];
            if((100-progresses[i]) % speeds[i]!=0){
                works[i]++;
            }
        }

        int prev=works[0];
        int cnt=1;
        for(int i=1; i<progresses.length; i++){
            if(works[i]>prev){
                answer.add(cnt);
                cnt=1;
                prev=works[i];
            }else{
                cnt++;
            }
        }
        answer.add(cnt);
        
        int[] answerArr = new int[answer.size()];
        for(int i=0; i<answer.size(); i++){
            answerArr[i] = answer.get(i);
        }
        
        return answerArr;
    }
}